frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -2521,6 +2526,78 @@ private void serveLicenseToSourceFileMapping(ResourceRequest request, ResourceRe
         }
     }
 
+    private void addLicenseToLinkedReleases(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
+        ProjectService.Iface client = thriftClients.makeProjectClient();
+        ComponentService.Iface componentClient = thriftClients.makeComponentClient();
+        LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
+        final User user = UserCacheHolder.getUserFromRequest(request);
+        final String projectId = request.getParameter(PROJECT_ID);
+        final JSONObject jsonResult = createJSONObject();
+
+        try {
+            Project project = client.getProjectById(projectId, user);
+            Set<String> releaseIds = CommonUtils.getNullToEmptyKeyset(project.getReleaseIdToUsage());
+            StringBuilder oneCLI = new StringBuilder();
+            StringBuilder multipleCLI = new StringBuilder();
+            StringBuilder noCLI = new StringBuilder();
+
+            for (String releaseId : releaseIds) {
+                Release release = componentClient.getReleaseById(releaseId, user);
+                List<Attachment> filteredAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+                Set<String> mainLicenses = release.getMainLicenseIds() == null ? new HashSet<String>() : release.getMainLicenseIds();
+                Set<String> otherLicenses = release.getOtherLicenseIds() == null ? new HashSet<String>() : release.getOtherLicenseIds();
+
+                if (filteredAttachments.size() == 1) {
+                    final Attachment attachment = filteredAttachments.get(0);
+                    final String attachmentName = attachment.getFilename();
+                    oneCLI.append(CommonUtils.nullToEmptyString(printName(release))).append(",");
+                    List<LicenseInfoParsingResult> licenseInfoResult = licenseInfoClient.getLicenseInfoForAttachment(release,
+                            attachment.getAttachmentContentId(), true, user);
+                    List<LicenseNameWithText> licenseWithTexts = licenseInfoResult.stream()
+                            .flatMap(result -> result.getLicenseInfo().getLicenseNamesWithTexts().stream())
```

Comment:
Have seen in some places null check for `result.getLicenseInfo()`. It would be good to add a check here as well. Same in line number 2564

Reply:
added the suggested change.